### PR TITLE
Issue 490 side reactions

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

Add side reactions for leading-order and full lead-acid models.
Fixes some of #490 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Key checklist:

- [x] No style issues: `$ flake8`
- [x] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks: 

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pybamm/models/full_battery_models/lead_acid/newman_tiedemann.py [hunk 1]
```diff
@@ -18,10 +18,11 @@ class NewmanTiedemann(BaseModel):
     **Extends:** :class:`pybamm.lead_acid.BaseModel`
     """
 
-    def __init__(self, options=None):
+    def __init__(self, options=None, name="Newman-Tiedemann model"):
         super().__init__(options)
-        self.name = "Newman-Tiedemann model"
+        self.name = name
```
Comment (Scottmar93) [inline pybamm/models/full_battery_models/lead_acid/newman_tiedemann.py]: Maybe set `self.name` in the base model so don't need this line in each?

File: pybamm/models/full_battery_models/lithium_ion/base_lithium_ion_model.py [hunk 2]
```diff
@@ -67,3 +67,15 @@ def set_standard_output_variables(self):
...
         )
+
+    def set_reactions(self):
+
+        # Should probably refactor as this is a bit clunky at the moment
+        # Maybe each reaction as a Reaction class so we can just list names of classes
+        icd = " interfacial current density"
+        self.reactions = {
+            "main": {
+                "Negative": {"s": 1, "aj": "Negative electrode" + icd},
```
Comment (Scottmar93) [inline pybamm/models/full_battery_models/lithium_ion/base_lithium_ion_model.py]: I can never remember what `s` is. Should it be (1-t+)?

File: pybamm/models/submodels/interface/base_interface.py [hunk 3]
```diff
@@ -100,14 +124,24 @@ def _get_standard_exchange_current_variables(self, j0):
             j0 = pybamm.Broadcast(j0, self.domain_for_broadcast)
 
         variables = {
-            self.domain + " electrode exchange current density": j0,
+            self.domain
+            + " electrode"
+            + self.reaction_name
```
Comment (Scottmar93) [inline pybamm/models/submodels/interface/base_interface.py]: :+1:

File: pybamm/models/full_battery_models/lithium_ion/base_lithium_ion_model.py (hunk 2 above)
Comment (tinosulzer) [reply pybamm/models/full_battery_models/lithium_ion/base_lithium_ion_model.py]: well spotted

**Pull request merged**